bls/src/main/java/tech/pegasys/teku/bls/impl/blst/BlstBLS12381.java

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.bls.impl.blst;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Collectors;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.apache.tuweni.bytes.Bytes48;
+import tech.pegasys.teku.bls.BatchSemiAggregate;
+import tech.pegasys.teku.bls.impl.BLS12381;
+import tech.pegasys.teku.bls.impl.KeyPair;
+import tech.pegasys.teku.bls.impl.PublicKey;
+import tech.pegasys.teku.bls.impl.Signature;
+import tech.pegasys.teku.bls.impl.blst.swig.BLST_ERROR;
+import tech.pegasys.teku.bls.impl.blst.swig.blst;
+import tech.pegasys.teku.bls.impl.blst.swig.p2;
+import tech.pegasys.teku.bls.impl.blst.swig.p2_affine;
+import tech.pegasys.teku.bls.impl.blst.swig.pairing;
+
+public class BlstBLS12381 implements BLS12381 {
+
+  public static BlstBLS12381 INSTANCE = new BlstBLS12381();
+
+  private static final int BATCH_RANDOM_BYTES = 8;
+
+  static {
+    try {
+      NativeUtils.loadLibraryFromJar("/" + System.mapLibraryName("jblst"));
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
```

Comment:
This depends on class loading and initialisation behaviour which has been very unpredictable in the past (maybe the JVM spec has improved to make it more predictable now - not sure).  It may be worth making `INSTANCE` be an `Optional` that only has a value if the library was loaded successfully:
```suggestion
  private static final Logger LOG = LogManager.getLogger();
  
  public static final Optional<BlstBLS12381> INSTANCE;

  private static final int BATCH_RANDOM_BYTES = 8;

  static {
    boolean libraryLoaded;
    try {
      NativeUtils.loadLibraryFromJar("/" + System.mapLibraryName("jblst"));
      libraryLoaded = true;
      LOG.info("Successfully loaded native BLS library");
    } catch (IOException e) {
      LOG.warn("Couldn't load native BLS library: " + e);
      libraryLoaded = false;
    }
    INSTANCE = libraryLoaded ? Optional.of(new BlstBLS12381()) : Optional.empty();
  }
```

Reply:
Yeah, you are right, probably throwing exception from static initializer is not the best choice. Initially I was assuming that this is a critical error, but now I think that it's pretty normal situation (e.g. on Raspberry PI or other platforms not supported by blst). 
See d17224f2 fix